core-service/src/main/java/com/crmfoodestablishment/coreservice/service/MenuServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.crmfoodestablishment.coreservice.service;
+
+import com.crmfoodestablishment.coreservice.dto.MenuDto;
+import com.crmfoodestablishment.coreservice.entity.Menu;
+import com.crmfoodestablishment.coreservice.mapper.MenuMapper;
+import com.crmfoodestablishment.coreservice.repository.MenuRepository;
+import lombok.RequiredArgsConstructor;
+import org.mapstruct.factory.Mappers;
+import org.springframework.stereotype.Service;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class MenuServiceImpl implements MenuService {
+
+    private final MenuRepository menuRepository;
+    private final MenuMapper menuMapper = Mappers.getMapper(MenuMapper.class);
+    private final static String MENU_NOT_FOUND_ERROR_MESSAGE = "Menu with uuid %s is not found";
+
+    @Override
+    public UUID addMenu(MenuDto menuDto) {
+        if (menuRepository.existsByName(menuDto.getName())) {
+            throw new IllegalArgumentException("Menu with name " + menuDto.getName() + " already exists");
+        }
+
+        Menu menuToSave = menuMapper.mapMenuDtoToMenu(menuDto);
```

Comment:
Does it really keeps relationship between Menu and Dish synchronized, does it really created Dish object for each CreateDishDTO in MenuDTO?